src/amo/components/HeroRecommendation/index.js

```diff
@@ -60,6 +62,41 @@ export const addParamsToHeroURL = ({
 };
 
 export class HeroRecommendationBase extends React.Component<InternalProps> {
+  callToActionURL: string;
+
+  static defaultProps = {
+    _tracking: tracking,
+  };
+
+  constructor(props: InternalProps) {
+    super(props);
+
+    const { shelfData } = props;
+    invariant(shelfData, 'The shelfData property is required');
+
+    const { addon, external } = shelfData;
+    this.callToActionURL = '';
+
+    if (addon) {
+      this.callToActionURL = addParamsToHeroURL({
```

Comment:
It would be safer to make `callToActionURL` a method rather than a property. As a property created only at construction time, the URL will go stale if the component gets updated with a new `shelfData` prop. In the current app, that probably won't happen (because components are recreated by the router) but it could easily happen with some changes.

Reply:
Good idea. I wasn't too keen on the property approach, but I'm not sure why I didn't think to make it a method instead. This is much better.